sycl/include/CL/sycl/intel/sub_group.hpp

```diff
@@ -17,11 +17,11 @@
 #include <CL/sycl/detail/spirv.hpp>
 #include <CL/sycl/detail/type_traits.hpp>
 #include <CL/sycl/id.hpp>
+#include <CL/sycl/intel/bit_cast.hpp>
```

Comment:
Strictly speaking, this means that our implementation of sub-groups extension depends on another our extension, while there is no documented dependency in the specification.

I'm not sure how important this is and would like to hear feedback from @Pennycook here.

Reply:
I agree that this could be surprising, especially if we later introduce controls to enable individual extensions.

I'm not sure this makes sense, but could we do something like:
- Move this implementation of `bit_cast` to `cl::sycl::detail`
- Expose `bit_cast` as part of `cl::sycl::intel` only when some macro like `__SYCL_INTEL_BIT_CAST__` is defined

That way our implementation can use `detail::bit_cast` wherever it wants to, including in other extensions.  But users won't see `sycl::intel::bit_cast` unless they're expecting to.

Reply:
I agree with John, let's put actual implementation into `detail` namespace and use it from both: sub-groups and this extension

Reply:
> Expose bit_cast as part of ```cl::sycl::intel``` only when some macro like \_\_SYCL_INTEL_BIT_CAST\_\_ is defined

So, it should be similar to the following in SYCL_INTEL_unnamed_kernel_lambda extension? We should introduce a new compiler flag like ```-fsycl-bit_cast``` and when enabled, the extension should define the macro \_\_SYCL_INTEL_BIT_CAST\_\_?

https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/UnnamedKernelLambda/SYCL_INTEL_unnamed_kernel_lambda.asciidoc: 
> In the Intel prototype implementation of this extension at GitHub link, the extension is enabled through the compiler flag ```-fsycl-unnamed-lambda```. When enabled, the extension defines the macro \_\_SYCL_INTEL_UNNAMED_LAMBDA\_\_.

Reply:
That makes sense to me.

That said...  I'd personally be okay with this PR being merged without the new compiler flag and macro, as long as the function appears in both `detail::` and `cl::sycl::intel`.  We could then revisit how to enable/disable the extension later.  What do you think @AlexeySachkov?

Reply:
I'm okay to merge this PR without adding macro and/or compiler option for now